From a06f449cee6152ce8f0a051593fceb82d26e4f16 Mon Sep 17 00:00:00 2001
From: Josh Boyer <jwboyer@redhat.com>
Date: Fri, 26 Oct 2012 12:29:49 -0400
Subject: [PATCH] EFI: Add in-kernel variable to determine if Secure Boot is enabled
Patch-mainline: Not yet, from Fedora 18 kernel
References: fate#314574
Target: SLE-11 SP3

There are a few cases where in-kernel functions may need to know if
Secure Boot is enabled.  The added capability check cannot be used as the
kernel can't drop it's own capabilites, so we add a global variable
similar to efi_enabled so they can determine if Secure Boot is enabled.

Signed-off-by: Josh Boyer <jwboyer@redhat.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
Automatically created from "patches.suse/0004-EFI-Add-in-kernel-variable-to-determine-if-Secure-Boot-is-enabled.patch" by xen-port-patches.py

--- sle11sp3.orig/arch/x86/kernel/setup-xen.c	2013-02-05 16:50:11.000000000 +0100
+++ sle11sp3/arch/x86/kernel/setup-xen.c	2013-02-05 16:50:15.000000000 +0100
@@ -1185,11 +1185,15 @@ void __init setup_arch(char **cmdline_p)
 	io_delay_init();
 
 #ifndef CONFIG_XEN
-	if (boot_params.secure_boot)
+	if (boot_params.secure_boot) {
 #else
-	if (0)
+	if (0) {
 #endif
 		secureboot_enable();
+#ifdef CONFIG_EFI
+		secure_boot_enabled = 1;
+#endif
+	}
 
 #ifdef CONFIG_ACPI
 	if (!is_initial_xendomain()) {
